src/main/java/io/vertx/kafka/admin/impl/KafkaAdminClientImpl.java

```diff
@@ -301,6 +307,41 @@ public void describeConsumerGroups(List<java.lang.String> groupIds, Handler<Asyn
     return promise.future();
   }
 
+  @Override
+  public void describeCluster(Handler<AsyncResult<ClusterDescription>> completionHandler) {
+    describeCluster().onComplete(completionHandler);
+  }
+
+  @Override
+  public Future<ClusterDescription> describeCluster() {
+    ContextInternal ctx = (ContextInternal) vertx.getOrCreateContext();
+    Promise<ClusterDescription> promise = ctx.promise();
+
+    DescribeClusterResult describeClusterResult = this.adminClient.describeCluster();
+    KafkaFuture.allOf(describeClusterResult.clusterId(), describeClusterResult.controller(), describeClusterResult.nodes()).whenComplete((r, ex) -> {
+      if (ex == null) {
+        try {
+          String clusterId = describeClusterResult.clusterId().get();
+          org.apache.kafka.common.Node rcontroller = describeClusterResult.controller().get();
+          Collection<org.apache.kafka.common.Node> rnodes = describeClusterResult.nodes().get();
+
+          Node controller = new Node(rcontroller.hasRack(), rcontroller.host(), rcontroller.id(), rcontroller.idString(), rcontroller.isEmpty(), rcontroller.port(), rcontroller.rack());
```

Comment:
There is an `Helper.from` method for doing this conversion.

Reply:
great. thanks for the tip.